test/e2e/virtual/apiexport/virtualworkspace_test.go

```diff
@@ -229,7 +213,200 @@ func TestAPIExportVirtualWorkspace(t *testing.T) {
+		if err != nil {
+			return false, err.Error()
+		}
+
+		if conditions.IsTrue(sheriffExport, apisv1alpha1.APIExportIdentityValid) {
+			identityHash = sheriffExport.Status.IdentityHash
+			return true, ""
+		}
+		condition := conditions.Get(sheriffExport, apisv1alpha1.APIExportIdentityValid)
+		if condition != nil {
+			return false, fmt.Sprintf("not done waiting for API Export condition status:%v - reason: %v - message: %v", condition.Status, condition.Reason, condition.Message)
+		}
+		return false, "not done waiting for APIExportIdentiy to be marked valid, no condition exists"
+
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "could not wait for APIExport to be valid with identity hash")
+
+	t.Logf("Found identity hash: %v", identityHash)
+	apifixtures.BindToExport(ctx, t, serviceProviderSherriffs, "wild.wild.west", consumerWorkspace, kcpClients)
+	setUpServiceProviderWithPermissionClaims(ctx, dynamicClients, kcpClients, kubeClusterClient, serviceProviderWorkspace, t, identityHash)
+
+	t.Logf("Create a sherriff that will be before the binding has occurred to validate controller re-labels")
+	apifixtures.CreateSheriff(ctx, t, dynamicClients, consumerWorkspace, "wild.wild.west", "in-vw-before")
+	bindConsumerToProvider(ctx, consumerWorkspace, serviceProviderWorkspace, t, kcpClients, identityHash)
+
+	t.Logf("Create a sherriff that should be retrieved")
+	// We need to sleep for now, because the controller does not exist to make this eventually conistant.
+	time.Sleep(5 * time.Second)
+	apifixtures.CreateSheriff(ctx, t, dynamicClients, consumerWorkspace, "wild.wild.west", "in-vw")
+
+	t.Logf("bind and create a Sheriff in: %v to prove that it would not be retrieved", consumerWorkspace2)
+	apifixtures.BindToExport(ctx, t, serviceProviderSherriffs, "wild.wild.west", consumerWorkspace2, kcpClients)
+	// Wait for the cache to be caught up in the admision.
+	apifixtures.CreateSheriff(ctx, t, dynamicClients, consumerWorkspace2, "wild.wild.west", "not-in-vw")
+
+	t.Logf("test that the admin user can use the virtual workspace to get cowboys")
+	apiExport, err := kcpClients.Cluster(serviceProviderWorkspace).ApisV1alpha1().APIExports().Get(ctx, "today-cowboys", metav1.GetOptions{})
+	require.NoError(t, err, "error getting APIExport")
+	require.Len(t, apiExport.Status.VirtualWorkspaces, 1, "unexpected virtual workspace URLs: %#v", apiExport.Status.VirtualWorkspaces)
+
+	apiExportVWCfg := rest.CopyConfig(cfg)
+	apiExportVWCfg.Host = apiExport.Status.VirtualWorkspaces[0].URL
+
+	t.Logf("Ensuring the appropriate core/v1 resources are available")
+	dynamicVWClients, err := dynamic.NewClusterForConfig(apiExportVWCfg)
+	require.NoError(t, err, "error creating dynamic cluster client for %q", apiExportVWCfg.Host)
+
+	t.Logf("Verify that two sherrifs are eventually returned")
+	framework.Eventually(t, func() (done bool, str string) {
```

Comment:
Please hold on to the sherrif objects that you created and assert that you get *those* two